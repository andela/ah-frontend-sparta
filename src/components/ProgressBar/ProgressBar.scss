@import '../../assets/scss/shared.scss';

.progressBar {
    position: relative;
    height: rem(3);
    margin-bottom: rem(10);
    width: rem(150);
    border-radius: rem(50);
    border: rem(1) solid #172439;
}

.filler {
    background-color: #172439;
    height: 100%;
    border-radius: inherit;
    transition: width .2 ease-in;
}
