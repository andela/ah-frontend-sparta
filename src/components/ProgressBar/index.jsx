import React from 'react';
import PropTypes from 'prop-types';

import './ProgressBar.scss';

const ProgressBar = ({ percentage }) => (
  <div className="progressBar">
    <Filler percentage={percentage} />
  </div>
);

const Filler = ({ percentage }) => (
  <div className="filler" style={{ width: `${percentage}%` }} />
);

const propTypesVal = {
  percentage: PropTypes.number,
};

const defaultPropsValidation = {
  percentage: 0,
};

ProgressBar.propTypes = propTypesVal;
Filler.propTypes = propTypesVal;

ProgressBar.defaultProps = defaultPropsValidation;
Filler.defaultProps = defaultPropsValidation;

export default ProgressBar;
