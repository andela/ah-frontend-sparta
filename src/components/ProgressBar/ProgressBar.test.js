import React from 'react';
import renderer from 'react-test-renderer';
import ProgressBar from './index';

test('snapshot test', () => {
  const wrapper = renderer.create(<ProgressBar />);
  expect(wrapper).toMatchSnapshot();
});
