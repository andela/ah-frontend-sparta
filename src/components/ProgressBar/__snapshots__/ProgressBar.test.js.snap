// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot test 1`] = `
<div
  className="progressBar"
>
  <div
    className="filler"
    style={
      Object {
        "width": "0%",
      }
    }
  />
</div>
`;
